File: Fair-Candy-Swap.js
/**
 * @param {number[]} aliceSizes
 * @param {number[]} bobSizes
 * @return {number[]}
 */
var fairCandySwap = function(A, B) {
    let sumA = A.reduce((acc, num) => acc + num, 0);
    let sumB = B.reduce((acc, num) => acc + num, 0);
    let diff = (sumA - sumB) / 2;
    
    let setB = new Set(B);
    
    for (let candyA of A) {
        if (setB.has(candyA - diff)) {
            return [candyA, candyA - diff];
        }
    }
    return [];
};

File: Fair-Candy-Swap.cpp
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& A, vector<int>& B) {
        int sumA = 0, sumB = 0;
        for (int a : A) sumA += a;
        for (int b : B) sumB += b;
        
        int diff = (sumA - sumB) / 2;
        unordered_set<int> setB(B.begin(), B.end());
        
        for (int a : A) {
            if (setB.count(a - diff)) {
                return {a, a - diff};
            }
        }
        
        return {};
    }
};
